bcgov/HOUS-permit-portal
# Add ability to initialize step code from H2K XML

[BPHH-434]
[BPHH-430]

## Description

Adds initial DB modelling for step code and the ability to initialize a new step code from H2K XML (parsed with Nokogiri)

## What type of PR is this? (check all applicable)


## Related Tickets & Documents

Implements [BPHH-434](feature/BPHH-434/initialize-step-code-from-h2K-xml) and [BPH-430](https://hous-bssb.atlassian.net/browse/BPHH-430).

## Steps to QA

1. Migrate DB
2. Download test H2K file (see Jira ticket `test.h2k`)
3. Open a rails console and run the following:

```
xml_file_path = 'path/to/test.h2k'
xml = Nokogiri::XML(File.read(xml_file_path))
StepCode::InitializeFromHot2000.new(xml:).call
```

This should create a new `StepCode` model with a single `data_entry` association that matches the data in the h2k file

## UI Accessibility Checklist

N/A

## General Checklist

- [ ] ✅ Provide tests for your changes where relevant **- will do this as part of [BPHH-546](https://hous-bssb.atlassian.net/browse/BPHH-546)**

## [optional] Are there any post-deployment tasks we need to perform?

N/A
